### PR TITLE
DISPATCH-1779: initial http stats for http1 and http2 adaptors

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Comment by grs** [comment]
@kgiusti @ganeshmurthy

**Review by ganeshmurthy — APPROVED** [review]

**Comment by jiridanek** [comment]
I was a bit concerned about the system_tests_http1_adaptor failing in GH Action with

    Exception: wait_port timeout on host None port 23906: [Errno 111] Connection refused

because that might've been caused by the cgroups network sandbox that the Actions enable, but there is the same test failing on Travis as well, where the sandbox is not used. So it's probably not environmental.

**Comment by grs** [comment]
@jiridanek that test is failing independent of this PR though I believe, and I think @kgiusti is looking into it? Or is this PR introducing a new failure?

**Comment by jiridanek** [comment]
@grs The logs look a lot like what's currently on the dev-protocol-adaptors-2 branch, so you're most likely right.

I admit I am not following the branch closely; my interest is in making sure the CI is not broken due to missing dependencies (as it was at one point), or due to insufficiently configured networking, as the GHA might be in the future. If the CI becomes irrelevant at any point, people will stop watching it.

**Comment by grs** [comment]
Agreed. (And it did catch one issue specific to this PR, which I fixed.)

File: src/adaptors/http1/http1_client.c [hunk 1]
```diff
@@ -722,6 +724,17 @@ static int _client_rx_headers_done_cb(h1_codec_request_state_t *hrs, bool has_bo
...
     qd_compose_insert_string(props, h1_codec_request_state_method(hrs));  // subject
     qd_compose_insert_string(props, hconn->client.reply_to_addr);   // reply-to
+    qd_compose_insert_null(props);                      // correlation-id
+    qd_compose_insert_null(props);                      // content-type
+    qd_compose_insert_null(props);                      // content-encoding
+    qd_compose_insert_null(props);                      // absolute-expiry-time
+    qd_compose_insert_null(props);                      // creation-time
+    qd_compose_insert_string(props, hconn->cfg.site);   // group-id
+
+    assert(hconn->client.reply_to_addr && strlen(hconn->client.reply_to_addr));  //remove me
```
Comment (kgiusti) [inline src/adaptors/http1/http1_client.c:734]: Is this intended for inclusion with this patch?

**Review by kgiusti — APPROVED** [review]
Looks fine.  Would it be possible to have a follow up patch with unit tests?  Thanks.

**Comment by kgiusti** [comment]
@jiridanek @grs 
The HTTP1 system tests are not passing for a variety of issues that have nothing to do with this patch.
I'm working towards resolving them.  This patch should land regardless.

File: src/adaptors/http1/http1_client.c (hunk 1 above)
Comment (grs) [reply src/adaptors/http1/http1_client.c:734]: Not the assert! I must have accidentally included that when merging conflicts in the rebase. Will remove.

**Comment by grs** [comment]
@kgiusti I have added a test for http2 stats and will add for http1 once the existing http1 tests are passing (so I can reuse your setup etc)

**Comment by ganeshmurthy** [comment]
@grs the http2 system tests you have added look good to me.

**Review by ganeshmurthy — APPROVED** [review]

**Comment by grs** [comment]
@jiridanek your concern was justified, there was indeed an issue that caused additional test failures. Now fixed though and seems to be similar to the errors on clean dev branch.

**Comment by grs** [comment]
@kgiusti @ganeshmurthy if there are no objections I will merge this to the dev-protocol-adapaters-2 branch on Monday.

**Comment by ganeshmurthy** [comment]
> @kgiusti @ganeshmurthy if there are no objections I will merge this to the dev-protocol-adapaters-2 branch on Monday.

Please go right ahead. Thanks.

**Review by kgiusti — APPROVED** [review]

**Pull request merged**